ts/kpt-functions/src/types.ts

```diff
@@ -206,6 +206,26 @@ export class Configs {
     return cm ? cm.get(key) : undefined;
   }
 
+  /**
+   * Detects if an unknown value has been provided to the config map
+   * @param expectedKeys The set of expecte keys for the function
+   * @returns Returns true if the config map is undefined or a value in the
+   * config map is not a recognized key. If all data values in the config map
+   * are known keys this will return false.
+   */
+  hasUnexpetedFunctionParameter(expectedKeys: string[]): boolean {
```

Comment:
When it finds unknown parameter, it should tell which is unknown.

Reply:
Updated the function to return a `string[]`